src/jdk.javadoc/share/classes/jdk/javadoc/internal/doclets/toolkit/ConstantsSummaryWriter.java

```diff
@@ -55,13 +55,10 @@ public interface ConstantsSummaryWriter {
     /**
      * Adds the given package name link to the constant content list.
      *
-     * @param pkg                    the {@link PackageElement} to index.
-     * @param writtenPackageHeaders  the set of package headers that have already
-     *                               been indexed, we want to index utmost once.
-     * @param content                the content to which the link will be added
+     * @param abbrevPackageName the abbreviated package name
+     * @param content       the content to which the link will be added
      */
-    void addLinkToPackageContent(PackageElement pkg, Set<PackageElement> writtenPackageHeaders,
-                                 Content content);
+    void addLinkToPackageContent(String abbrevPackageName, Content content);
```

Comment:
It feels strange to downgrade a package representation from PackageElement to String. Especially considering that this interface expresses other program elements through javax.lang.model.

Reply:
Agreed it may seem strange.  Two points:
1. There is no representation for a package that ignores the enclosing module
2. These are not packages so much as headings, for the purpose of provides a "Contents" list at the head of the page, for ease of navigation.